tests/unit/defrag.cpp

```diff
@@ -0,0 +1,376 @@
+	std::sort(shadow_elems.begin(), shadow_elems.begin() + (RECORDS_IN_BLOB - RECORDS_TO_REMOVE));
+
+	// TODO: need to check that we use view over base
+	BOOST_REQUIRE_EQUAL(datasort(wrapper, {0}), 0);
+
+	iterator_private priv(shadow_elems);
+	BOOST_REQUIRE_EQUAL(iterate(wrapper, priv), 0);
+	BOOST_REQUIRE_EQUAL(priv.number_checked, TOTAL_RECORDS - RECORDS_TO_REMOVE);
+}
+
+
+/**
+ * 1) Make two bases with 100 records each.
+ *    State: data.0(unsorted, 100 records), data.1(unsorted, 100 records)
+ * 2) Defrag first base
+ *    State: data.0(sorted, 100 records), data.1(unsorted, 100 records)
+ * 3) Remove a half from each base
+ *    State: data.0(sorted, 50 records), data.1(unsorted, 50 records)
+ * 4) Defrag two bases with view on first base
+ *    State: data.0(sorted, 100 records)
+ * 5) Check that result base contains 100 records
+ */
+BOOST_AUTO_TEST_CASE(merge_sorted_and_unsorted_bases) {
+	const size_t TOTAL_RECORDS = 2 * RECORDS_IN_BLOB;
+	const size_t RECORDS_TO_REMOVE_IN_BASE = RECORDS_IN_BLOB / 2;
+
+	auto config = initialize_eblob_config_for_defrag();
+	eblob_wrapper wrapper(config.get());
+	wrapper.start();
+	BOOST_REQUIRE(wrapper.get() != nullptr);
+
+	item_generator generator(wrapper);
+	std::vector<item_t> shadow_elems;
+
+	fill_eblob(wrapper, shadow_elems, generator, TOTAL_RECORDS);
+
+	// Sort first base
+	BOOST_REQUIRE_EQUAL(datasort(wrapper, {0}), 0);
+
+	// Remove a half from first base
+	for (size_t index = 0; index != RECORDS_TO_REMOVE_IN_BASE; ++index) {
+		BOOST_REQUIRE_EQUAL(wrapper.remove_item(shadow_elems[index]), 0);
+	}
+
+	// Remove a half from second base
+	for (size_t index = 0; index != RECORDS_TO_REMOVE_IN_BASE; ++index) {
+		BOOST_REQUIRE_EQUAL(wrapper.remove_item(shadow_elems[RECORDS_IN_BLOB + index]), 0);
+	}
+
+	// Defrag two bases
+	BOOST_REQUIRE_EQUAL(datasort(wrapper, {0, 1}), 0);
+	shadow_elems.erase(shadow_elems.begin() + RECORDS_IN_BLOB, shadow_elems.begin() + RECORDS_IN_BLOB + RECORDS_TO_REMOVE_IN_BASE);
+	shadow_elems.erase(shadow_elems.begin(), shadow_elems.begin() + (RECORDS_IN_BLOB - RECORDS_TO_REMOVE_IN_BASE));
+	std::sort(shadow_elems.begin(), shadow_elems.end());
+
+	iterator_private priv(shadow_elems);
+	BOOST_REQUIRE_EQUAL(iterate(wrapper, priv), 0);
+	BOOST_REQUIRE_EQUAL(priv.number_checked, TOTAL_RECORDS - 2 * RECORDS_TO_REMOVE_IN_BASE);
+}
+
+
+/**
+ *  1) Make two bases with 100 records each.
+ *     State: data.0(unsorted, 100 record), data.1(unsorted, 100 records)
+ *  2) Defrag two bases separately
+ *     State: data.0(sorted, 100 records), data.1(sorted, 100 records)
+ *  3) Remove a half from each base
+ *     State: data.0(sorted, 50 records), data.1(sorted, 50 records)
+ *  4) Defrag two bases with view on first base
+ *     State: data.0(sorted, 100 records)
+ *  5) Check that result base contains 100 records
+ */
+BOOST_AUTO_TEST_CASE(merge_sorted_and_sorted_bases) {
+	const size_t TOTAL_RECORDS = 2 * RECORDS_IN_BLOB;
```

Comment:
Here and after we can create config before that constants and do
```suggestion
	const size_t TOTAL_RECORDS = 2 * config.get().records_in_blob;
```

Reply:
done